src/testsets.jl

```diff
@@ -15,22 +33,41 @@ flatted upon `finish`. See `bin/reporttests.jl` for an example of this use.
 and this is not recommended or supported.
 
 A `ReportingTestSet` has the `description` and `results` fields as per a
-`DefaultTestSet`, and has an additional `properties` field which is used
-to record properties to be inserted into the report.
+`DefaultTestSet`, and has the following additional fields:
+
+- `properties`: a dictionary which is used to record properties to be
+    inserted into the report.
+- `start_time::DateTime`: the start date and time of the testing (UTC).
+- `time_taken`: the time taken in seconds to run the `TestSet`.
+- `last_record_time`: the time when `record` was last called.
 
 See also: [`flatten_results!`](@ref), [`recordproperty`](@ref), [`report`](@ref)
 """
 mutable struct ReportingTestSet <: AbstractTestSet
     description::String
     results::Vector
     properties::Dict{String, Any}
+    start_time::DateTime
+    time_taken::Float64
```

Comment:
It is worth considering having this be of the `Milisecond` type.
Which is backed by an integer.
That way we can avoid and floating point loss of precision.
Though I guess precision should be pretty much perfect for the small numbers we are expecting.

Still Millisecond is convenient since that is the default type returned when you subtract DateTimes iirc

Reply:
This sounds good, will need something to format it for XML (needs to be decimal value in seconds) but that's easy to do!